wordpress-mobile/WordPress-iOS
# Feature/5310 reader search autocomplete

Refs #5310 
This PR adds a suggestions feature to the reader search bar.  
- Previously searched phrases appear below the search bar, sorted by most recently searched, and filtered as the user types. 
- Individual searches can be deleted by a swipe to delete gesture.
- All saved searches can be cleared via the "Clear saved searches" button.
- Tapping a phrase loads its content.

Examples:
https://cloudup.com/ihR0zIz7tC7
@jancavan, could you take a peek at the design? The iPad version in particular. I think the functionality is what we want but design-wise it could use a little TLC.

To test: 
Prep: Search for a few different topics in the reader. Some of them should start with the same characters, some should not.

1. Tap into the search form and confirm you see the full list of saved searches. 
2. Start typing your search phrase. You should see the saved searches filtered. 
3. Tap a saved search phrase. Content for that search phrase should be fetched/loaded. 
4. Swipe to delete a single search term. 
5. Clear all search terms. 
6. Confirm the list of saved searches is hidden when the search bar resigns first responder.  You can perform a search or just tap anywhere outside of the search bar / search suggestions to test. 

Needs review: @jleandroperez would you mind taking a look at another search PR? 

cc @nbradbury
